File: src/main/java/org/svomz/apps/finances/domain/model/AccountRepository.java
package org.svomz.apps.finances.domain.model;

import java.util.List;
import java.util.Optional;

/**
 * Created by eric on 28/02/16.
 */
public interface AccountRepository {

  List<Account> findAll();

  Optional<Account> find(String id);

  AccountId nextIdentity();

  Account create(Account account);
}

File: core/src/main/scala/org/svomz/apps/finances/core/domain/model/AccountRepository.scala
package org.svomz.apps.finances.core.domain.model

import scala.concurrent.Future

trait AccountRepository {

  def update(account: Account): Future[Account]

  def delete(account: Account): Future[Account]

  def create(account: Account): Future[Account]

  def fetch(no: AccountId): Future[Option[Account]]

  def all: Future[List[Account]]

  def nextIdentity: Future[AccountId]

}
